MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp

```diff
@@ -612,6 +619,93 @@ void EnggDiffractionViewQtGUI::plotReplacingWindow(const std::string &wsName) {
   m_presenter->notify(IEnggDiffractionPresenter::LogMsg);
 }
 
+void EnggDiffractionViewQtGUI::plotVanCurvesCalibOutput() {
+  std::string pyCode =
+      "van_curves_ws = workspace(\"engggui_vanadium_curves_ws\")\n"
+      "plot(van_curves_ws, [0, 1, 2])";
+
+  std::string status =
+      runPythonCode(QString::fromStdString(pyCode), false).toStdString();
+
+  m_logMsgs.push_back(
+      "Plotted output calibration vanadium curves, with status string " +
+      status);
+  m_presenter->notify(IEnggDiffractionPresenter::LogMsg);
+}
+
+void EnggDiffractionViewQtGUI::plotDifcZeroCalibOutput(
+    std::vector<double> &difc, std::vector<double> &tzero,
+    std::string &specNo) {
+
+  size_t bank1 = size_t(0);
+  size_t bank2 = size_t(1);
+  std::string pyRange;
+
+  // sets the range to plot appropriate graph for the particular bank
+  if (specNo == "North") {
+    pyRange = "1, 2";
+  } else if (specNo == "South") {
+    pyRange = "2, 3";
+    bank2 = size_t(0);
+  } else {
+    pyRange = "1, 3";
+  }
+
+  std::string pyCode =
+      "for i in range(" + pyRange +
+      "):\n"
+      " bank_ws = workspace(\"engggui_calibration_bank_\" + str(i))\n"
+
+      " xVal = []\n"
+      " yVal = []\n"
+      " y2Val = []\n"
+
+      " for irow in range(0, bank_ws.rowCount()):\n"
+      "  xVal.append(bank_ws.cell(irow, 0))\n"
+      "  yVal.append(bank_ws.cell(irow, 5))\n"
+
+      "  if (i == 1):\n"
+      "   difc=" +
+      boost::lexical_cast<std::string>(difc[bank1]) + "\n" + "   tzero=" +
+      boost::lexical_cast<std::string>(tzero[bank1]) + "\n" + "  else:\n"
+
+                                                              "   difc=" +
+      boost::lexical_cast<std::string>(difc[bank2]) + "\n" + "   tzero=" +
+      boost::lexical_cast<std::string>(tzero[bank2]) + "\n" +
```

Comment:
There's a few places in this method where the code is assuming bank1 and bank2. Later on this could be refactored into a function that processes one bank, which would be called for each of the banks passed/requested.